### PR TITLE
Fix automatic mut for non-streams

### LINKED ISSUE #11086: Mutable bindings throw exception for non-stream inputs
Failing test case here: https://github.com/emberjs/ember.js/compare/master...ef4:mutable-boom

### PR DESCRIPTION
**Pull request by ef4**
Fixes https://github.com/emberjs/ember.js/issues/11086

### CODE REVIEW

File: packages/ember-htmlbars/tests/integration/mutable_binding_test.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ QUnit.test('using a string value through middle tier does not trigger assertion'
 
   runAppend(view);
 
-  assert.strictEqual(bottom.attrs.stuff.value, 'foo', "precond - the data propagated");
+  assert.strictEqual(bottom.attrs.stuff, 'foo', "precond - the data propagated");
```
Comment (ef4) [inline packages/ember-htmlbars/tests/integration/mutable_binding_test.js]: This test was trolling me. It was incorrectly asserting that the non-stream value became a stream.

Comment (rwjblue) [inline packages/ember-htmlbars/tests/integration/mutable_binding_test.js]: Not sure I follow.  the `bottom.attrs.stuff.value` was not asserting that it was a stream, but that it was a `MUTABLE_CELL`.  I believe that in this case, the assertion was correct.

The invoking site was `{{bottom-mut stuff=attrs.value}}`, and with "auto-mut" mode that should result in the same thing as `{{bottom-mut stuff=(mut attrs.value)}}` which would be a `MUTABLE_CELL`.

Comment (ef4) [inline packages/ember-htmlbars/tests/integration/mutable_binding_test.js]: I see. You're right, we need to give the components mutable cells if that's what they're expecting, even thought the original source is a constant that they can't really change.

**Comment by ef4** [comment]
Second attempt pushed. This one upgrades any non-streams to streams that look mutable to the consumer but don't really go anywhere higher.

**Comment by mixonic** [comment]
lgtm.

**Pull request merged**